virttest/guest_agent.py

```diff
@@ -497,19 +502,56 @@ def shutdown(self, mode=SHUTDOWN_MODE_POWERDOWN):
         return True
 
     @error_context.context_aware
-    def sync(self):
+    def sync(self, sync_mode="guest-sync"):
         """
-        Sync guest agent with cmd 'guest-sync'.
+        Sync guest agent with cmd 'guest-sync' or 'guest-sync-delimited'.
         """
-        cmd = "guest-sync"
+        cmd = sync_mode
         if not self._has_command(cmd):
             return
 
-        synced = self._sync()
+        synced = self._sync(sync_mode)
         if not synced:
             raise VAgentSyncError(self.vm.name)
 
     @error_context.context_aware
+    def set_user_password(self, password, crypted=False, username="root"):
+        """
+        Set the new password for the user
+        """
+        cmd = "guest-set-user-password"
+        if not self._has_command(cmd):
+            return False
+
+        if crypted:
+            openssl_cmd = "openssl passwd -crypt %s" % password
+            password = process.system_output(openssl_cmd).strip('\n')
+
+        args = {"crypted": crypted, "username": username,
+                "password": base64.b64encode(password)}
+        return self.cmd(cmd=cmd, args=args)
+
+    @error_context.context_aware
+    def get_vcpus(self):
+        """
+        Get the vcpus infomation
+        """
+        cmd = "guest-get-vcpus"
+        if not self._has_command(cmd):
+            return False
```

Comment:
I handle the error in the upper layer (in tp-qemu).  In my opinion, for API, it's better to just give a result.   I add the error handler in https://github.com/autotest/tp-qemu/pull/623